### DISCUSSION TITLE
Using VaporFile and VaporImage on standard hosting

### DISCUSSION THREAD

**Discussion by bengravesdev** [laravel/nova-issues#5100] [Q&A]
I host my sites on Digital Ocean and deploy them using Forge.

It would be really nice to be able to use VaporFile and VaporImage to upload multigigabyte files directly to S3 regardless of whether they're hosted on Vapor.

I tried to use VaporFile but got a 404 in the console as it tried to find /vapor/signed-storage-url

**Comment by crynobone** [comment]
You need to depends on https://github.com/laravel/vapor-core

**Reply by bengravesdev** [reply]
> You need to depends on https://github.com/laravel/vapor-core

I tried installing laravel/vapor-core using composer but get the following error when attempting to upload a file in Nova:

production.ERROR: Unable to issue signed URL. Missing environment variables: AWS_BUCKET, AWS_DEFAULT_REGION, AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY

All those settings are in my .env file and work fine with the standard File field when S3 is specified as the disk.

**Reply by crynobone** [reply]
[SignedStorageUrlController](https://github.com/laravel/vapor-core/blob/2.0/src/Http/Controllers/SignedStorageUrlController.php) access them using `$_ENV` and not `env()`, you should be able to extend the class and override it via IoC. See https://github.com/nova-kit/nova-on-vapor/blob/563a8b2273b34573efb83b111c40b76607fd640a/src/LaravelServiceProvider.php#L42-L45 for example

**Reply by devoNOTbevo** [reply]
Hi there! Came across this while having the same issue. Is `$_ENV` not supported in Forge?
